File: app/src/main/java/com/niuda/a3jidi/laok/booklook/view/activity/MainActivity.java
package com.niuda.a3jidi.laok.booklook.view.activity;

import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.niuda.a3jidi.laok.R;
import com.niuda.a3jidi.laok.booklook.model.Book;
import com.niuda.a3jidi.laok.booklook.model.IBookView;
import com.niuda.a3jidi.laok.booklook.presenter.BookPresenter;
import com.niuda.a3jidi.lib_base.base.base.BaseActivity;

import butterknife.BindView;

public class MainActivity extends BaseActivity{
    @BindView(R.id.textview)
    TextView mTextview;
    @BindView(R.id.button)
    Button mButton;
    private BookPresenter mBookPresenter = new BookPresenter(this);


    @Override
    protected int initLayout() {
        return R.layout.activity_main;
    }

    @Override
    protected void initView() {
        mBookPresenter.onCreate();
        mBookPresenter.attachView(mBookView);
        mButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Log.i("TAG","sendRequest");
                mBookPresenter.getSearchBook("孙子兵法", null, 1, 10);
            }
        });
    }


    private IBookView mBookView = new IBookView() {
        @Override
        public void onSuccess(Book book) {
            mButton.setVisibility(View.GONE);
            mTextview.setText(book.toString());
        }

        @Override
        public void onError(String result) {
            Toast.makeText(MainActivity.this, result, Toast.LENGTH_SHORT).show();
        }
    };


    @Override
    protected void onDestroy() {
        super.onDestroy();
        mBookPresenter.onStop();
    }

}

File: app/src/main/java/com/niuda/a3jidi/laok/booklook/view/activity/MainActivity.kt
package com.niuda.a3jidi.laok.booklook.view.activity

import android.content.Context
import android.os.Bundle
import com.niuda.a3jidi.laok.R
import com.niuda.a3jidi.laok.booklook.MyApplication
import com.niuda.a3jidi.laok.booklook.contract.BookContract
import com.niuda.a3jidi.laok.booklook.di.components.MainComponent
import com.niuda.a3jidi.laok.booklook.model.pojo.Book
import com.niuda.a3jidi.laok.booklook.presenter.BookPresenter
import com.niuda.a3jidi.lib_base.base.base.BaseActivity
import kotlinx.android.synthetic.main.activity_main.*
import javax.inject.Inject

//@TranslucentStatusBar
class MainActivity : BaseActivity(),BookContract.BookView {

    var mMainComponent : MainComponent? = null;

    override fun getViewContext(): Context = this

    override fun layoutResId(): Int  = R.layout.activity_main

    @Inject lateinit var presenter: BookPresenter

    override fun onCreated(savedInstanceState: Bundle?) {
        mMainComponent = DaggerMainComponent.builder().userAppComponent(MyApplication.getAppComponent())
                .build()
                .inject(this)
        initToolbar(true).setPageTitle("首页")
        initView()
    }


    private fun initView() {
        button_main.setOnClickListener {
            presenter.getBookView()
        }
    }


    override fun Success(it: Book) {
        textview.text = it.toString()
    }

}
